internal/resource/resource.go

```diff
@@ -348,6 +352,15 @@ func NewResource(ctx context.Context, renv *readiness.Env, slice *apiv1.Resource
 	return res, nil
 }
 
+// Less returns true when r < than.
+// Used to establish determinstic ordering for conflicting resources.
+func (r *Resource) Less(than *Resource) bool {
+	if r.Manifest == nil || than.Manifest == nil {
+		return true
+	}
+	return r.Manifest.Manifest < than.Manifest.Manifest
```

Comment:
Is the string representation deterministic?

Reply:
Yes, since we only care about determinism within the context of a particular synthesis. So if the synthesizer moves around the order of some properties in the next synthesis it doesn't matter from this func's perspective.